### PR TITLE
[FEAT] Simulation paths

### PR DESCRIPTION
**Pull request by elephaint**
### Summary

Adds `NeuralForecast.simulate()`, a method for generating temporally correlated sample paths from fitted models. Unlike `predict(level=...)` which produces independent per-step intervals, `simulate()` generates full joint trajectories that preserve temporal correlation, enabling scenario analysis.

### New API

```python
nf.simulate(
    n_paths=100,          # number of sample paths
    seed=42,              # reproducibility
    method="gaussian_copula",  # or "schaake_shuffle"
)
# Returns DataFrame: [unique_id, ds, sample_id, model_1, model_2, ...]
```

### Simulation Methods

| Method | Description |
|--------|-------------|
| `gaussian_copula` (default) | AR(1) Gaussian copula with Toeplitz correlation and IQF marginals ([Baron et al., 2025](https://arxiv.org/abs/2510.02224)) |
| `schaake_shuffle` | Nonparametric — independent marginal samples reordered by historical rank templates ([Clark et al., 2004](https://journals.ametsoc.org/view/journals/hydr/5/1/1525-7541_2004_005_0243_tssamf_2_0_co_2.xml)) |

We don't (yet) support the (baseline) "roll-forward" autoregressive simulation. This is for a number of reasons:
- The roll-forward is very slow (requires H predict calls instead of 1). To do it efficiently, requires substantial rewrite in base_model.py
- The roll-forward suffers from noise / variance explosion (I think in Baron et al. they call it snowball effect)

### Documentation
**`nbs/docs/tutorials/simulation.ipynb`** — Tutorial demonstrating different loss types, decision-making example, and a side-by-side comparison of Gaussian copula vs. Schaake shuffle.

### Open issues (I suggest to leave open for now)
- `h > self.h` (simulating with horizon larger than train horizon) is not yet supported. This is intentional, as it requires substantial more work (and testing) to feed back the generated sample paths autoregressively for both recurrent and direct models. See also why the roll-forward approach hasn't been implemented.
- explainability also not supported for simulated paths.

### CODE REVIEW

File: neuralforecast/common/_base_model.py [hunk 1]
```diff
@@ -2098,6 +2098,122 @@ def predict(
 
         return fcsts
 
+    def simulate(
+        self,
+        dataset,
+        n_paths=500,
```
Comment (marcopeix) [inline neuralforecast/common/_base_model.py:2104]: Isn't this a bit too much? Maybe 100 is a more reasonable default?

File: tests/test_simulate.py [hunk 2]
```diff
@@ -0,0 +1,814 @@
...
+import pytest
+import torch
+
+
+def _get_model_cols(df):
+    """Get model columns (everything except unique_id, ds, sample_id)."""
+    return [c for c in df.columns if c not in ("unique_id", "ds", "sample_id")]
+
+
+class TestSimulation:
```
Comment (marcopeix) [inline tests/test_simulate.py:14]: Would be good to have a test when we have `futr_exog`.

**Review by marcopeix — CHANGES_REQUESTED** [review]
Very nice, and the tutorial is great too! Some minor comments.

File: neuralforecast/common/_base_model.py (hunk 1 above)
Comment (elephaint) [reply neuralforecast/common/_base_model.py:2104]: Good point, yeah
